crates/net/network/src/transactions/validation.rs

```diff
@@ -0,0 +1,364 @@
+
+/// L1 Network.
+#[derive(Debug, Display, Default)]
+pub struct LayerOne;
+
+impl ValidateTx68 for LayerOne {
+    fn should_fetch(&self, ty: u8, hash: TxHash, size: usize) -> ValidationOutcome {
+        // 1. checks if tx type is valid value for this network
+        let tx_type = match TxType::try_from(ty) {
+            Ok(ty) => ty,
+            Err(_) => {
+                debug!(target: "net::eth-wire",
+                    ty=ty,
+                    size=size,
+                    hash=%hash,
+                    network=%Self,
+                    "invalid tx type in eth68 announcement"
+                );
+
+                return ValidationOutcome::ReportPeer
+            }
+        };
+        // 2. checks if tx's encoded length is within limits for this network
+        if size > tx_type.max_encoded_tx_length() || size < tx_type.min_encoded_tx_length() {
+            // todo: min valid tx encoded length on layer 1? signature length?
+            debug!(target: "net::eth-wire",
+                ty=ty,
+                size=size,
+                hash=%hash,
+                max_encoded_tx_length=tx_type.max_encoded_tx_length(),
+                network=%Self,
+                "invalid tx size in eth68 announcement"
+            );
+
+            return ValidationOutcome::Ignore
+        }
+
+        ValidationOutcome::Fetch
+    }
+}
+
+impl FilterAnnouncement68 for LayerOne
+where
+    Self: ValidateTx68,
+{
+    fn filter_valid_entries(&self, msg: &mut NewPooledTransactionHashes68) -> FilterOutcome {
+        trace!(target: "net::tx::validation",
+            types=?msg.types,
+            sizes=?msg.sizes,
+            hashes=?msg.hashes,
+            network=%Self,
+            "validating eth68 announcement data.."
+        );
+
+        // 1. checks if the announcement is empty
+        if msg.hashes.is_empty() {
+            debug!(target: "net::tx",
+                network=%Self,
+                "empty announcement"
+            );
+            return FilterOutcome::ReportPeer
+        }
+
+        let mut should_report_peer = false;
+        let mut indices_to_remove = vec![];
+
+        for (i, (&ty, &hash, &size)) in izip!(&msg.types, &msg.hashes, &msg.sizes).enumerate() {
+            match self.should_fetch(ty, hash, size) {
+                ValidationOutcome::Fetch => (),
+                ValidationOutcome::Ignore => indices_to_remove.push(i),
+                ValidationOutcome::ReportPeer => {
+                    indices_to_remove.push(i);
+                    should_report_peer = true;
+                }
+            }
+        }
+
+        for (i, &index) in indices_to_remove.iter().enumerate() {
+            msg.hashes.remove(index - i);
+            msg.types.remove(index - i);
+            msg.sizes.remove(index - i);
+        }
+
+        let mut hashes_sorted = msg.hashes.clone();
+        hashes_sorted.sort();
+
+        for hash_pair in hashes_sorted.windows(2) {
```

Comment:
yeah but it could be hashes from two different announcements. though ab this too, that maybe we can check against backup peers, but that won't reveal peers that send messages all full of one hash. maybe we still want to catch that spam.